config/resolve_assume_role_test.go

```diff
@@ -0,0 +1,201 @@
+package config
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/internal/awstesting"
+)
+
+func TestAssumeRole(t *testing.T) {
+	restoreEnv := initConfigTestEnv()
+	defer awstesting.PopEnv(restoreEnv)
+
+	os.Setenv("AWS_REGION", "us-east-1")
+	os.Setenv("AWS_SHARED_CREDENTIALS_FILE", testConfigFilename)
+	os.Setenv("AWS_PROFILE", "assume_role_w_creds")
+
+	client := mockHttpClient(func(r *http.Request) (*http.Response, error) {
+		return &http.Response{
+			StatusCode: 200,
+			Body: ioutil.NopCloser(bytes.NewReader([]byte(fmt.Sprintf(assumeRoleRespMsg,
+				time.Now().Add(15*time.Minute).Format("2006-01-02T15:04:05Z"))))),
+		}, nil
+	})
+
+	config, err := LoadDefaultConfig(WithHTTPClient{client})
+	if err != nil {
+		t.Fatalf("expect no error, got %v", err)
+	}
+
+	creds, err := config.Credentials.Retrieve(context.Background())
+	if err != nil {
+		t.Fatalf("expect no error, got %v", err)
+	}
+	if e, a := "AKID", creds.AccessKeyID; e != a {
+		t.Errorf("expect %v, got %v", e, a)
+	}
+	if e, a := "SECRET", creds.SecretAccessKey; e != a {
+		t.Errorf("expect %v, got %v", e, a)
+	}
+	if e, a := "SESSION_TOKEN", creds.SessionToken; e != a {
+		t.Errorf("expect %v, got %v", e, a)
+	}
+	if e, a := "AssumeRoleProvider", creds.Source; !strings.Contains(a, e) {
+		t.Errorf("expect %v, to be in %v", e, a)
+	}
+}
+
+func TestAssumeRole_WithMFA(t *testing.T) {
+	restoreEnv := initConfigTestEnv()
+	defer awstesting.PopEnv(restoreEnv)
+
+	os.Setenv("AWS_REGION", "us-east-1")
+	os.Setenv("AWS_SHARED_CREDENTIALS_FILE", testConfigFilename)
```

Comment:
Will be helpful to rebase this PR on top of the ci with windows change to make sure this works for windows.